src/staking/phase1/index.ts

```diff
@@ -0,0 +1,349 @@
+   */
+  public createWithdrawEarlyUnbondedTransaction = (
+    stakingParams: Phase1Params,
+    delegation: Phase1Delegation,
+    unbondingTx: Transaction,
+    feeRate: number,
+  ): PsbtTransactionResult => {
+    const { stakingTx } = this.validateAndDecodeDelegaitonInputs(
+      delegation, stakingParams,
+    );
+    // Build scripts
+    const scripts = buildScripts(
+      stakingParams,
+      delegation.finalityProviderPkHex,
+      stakingTx.timelock,
+      delegation.stakerPkHex,
+    );
+
+    // Create the withdraw early unbonded transaction
+    try {
+      return withdrawEarlyUnbondedTransaction(
+        scripts,
+        unbondingTx,
+        this.stakerInfo.address,
+        this.network,
+        feeRate,
+      );  
+    } catch (error) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.BUILD_TRANSACTION_FAILURE,
+        "Cannot build unsigned withdraw early unbonded transaction",
+      );
+    }
+  }
+
+  /**
+   * Create a withdraw timelock unbonded transaction for Phase 1 of the Babylon Staking protocol.
+   * 
+   * @param {Phase1Params} stakingParams - The staking parameters for Phase 1.
+   * @param {Phase1Delegation} delegation - The delegation to withdraw from.
+   * @param {number} feeRate - The fee rate for the transaction in satoshis per byte.
+   * 
+   * @returns {PsbtTransactionResult} - An object containing the unsigned psbt and fee
+   * 
+   * @throws {StakingError} - If the delegation is invalid or the transaction cannot be built
+   */
+  public createWithdrawTimelockUnbondedTransaction = (
+    stakingParams: Phase1Params,
+    delegation: Phase1Delegation,
+    feeRate: number,
+  ): PsbtTransactionResult => {
+    const { stakingTx, btcStakingTx } = this.validateAndDecodeDelegaitonInputs(
+      delegation, stakingParams,
+    );
+    // Build scripts
+    const scripts = buildScripts(
+      stakingParams,
+      delegation.finalityProviderPkHex,
+      stakingTx.timelock,
+      delegation.stakerPkHex,
+    );
+
+    // Create the timelock unbonded transaction
+    try {
+      return withdrawTimelockUnbondedTransaction(
+        scripts,
+        btcStakingTx,
+        this.stakerInfo.address,
+        this.network,
+        feeRate,
+        stakingTx.outputIndex,
+      );  
+    } catch (error) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.BUILD_TRANSACTION_FAILURE,
+        "Cannot build unsigned timelock unbonded transaction",
+      );
+    }
+  }
+
+  validateAndDecodeDelegaitonInputs = (
+    delegation: Phase1Delegation, stakingParams: Phase1Params,
+  ) => {
+    const stakingTx = delegation.stakingTx;
+    if (stakingTx.startHeight < stakingParams.activationHeight) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_INPUT,
+        "Staking transaction start height cannot be less than activation height",
+      );
+    } else if (
```

Comment:
I recommend we just have `if` blocks instead of `if/else if` as all of them throw an error eitherway.